src/lib/Lib/AbsoluteCompilationDatabase.cpp

```diff
@@ -51,14 +51,22 @@ static
 std::vector<std::string>
 adjustCommandLine(
     const std::vector<std::string>& cmdline,
-    const std::vector<std::string>& additional_defines)
+    const std::vector<std::string>& additional_defines,
+    std::unordered_map<std::string, std::vector<std::string>> const& includePathsByCompiler)
 {
     std::vector<std::string> new_cmdline;
     std::vector<std::string> discarded_cmdline;
     llvm::opt::InputArgList args;
     StringRef driver_mode;
-    if(! cmdline.empty())
+
+    std::vector<std::string> systemIncludePaths;
+
+    if( ! cmdline.empty())
     {
+        if (auto it = includePathsByCompiler.find(cmdline[0]); it != includePathsByCompiler.end()) {
```

Comment:
Can `cmdline[0]` ever be different since CMake only accepts one toolchain at a time? I've never seen a case where it's different. 

In any case, it's probably fine to generalize this and parse this information for "all compilers we see" even if it's technically always the same compiler. Compilers can still technically be different according to the file format in compile_commands.json.